tests/unit/swagger/definitionsGeneration/definitions.spec.ts

```diff
@@ -1169,13 +1188,13 @@ describe('Definition generation', () => {
                     format: undefined,
                   },
                   indexedResponseObject: {
-                    $ref: '#/definitions/Record_id.any_',
+                    $ref: '#/definitions/Record_id._myProp1-string__',
```

Comment:
Previous name generation worked well only if we were not in a nested mapped type. Fixed to check&handle name conflicts